Lombiq.ChartJs.Samples/Controllers/SampleController.cs

```diff
@@ -0,0 +1,214 @@
+                                index.ContentField == nameof(ExpensePart.Amount) &&
+                                index.Numeric != null)
+                            .ListAsync())
+                        .Select(index => decimal.ToDouble(index.Numeric ?? 0m))
+                        .Sum(),
+                    },
+                },
+            },
+            Options = new
+            {
+                Animation = new
+                {
+                    Duration = 0,
+                },
+                Layout = new
+                {
+                    AutoPadding = false,
+                    Padding = 0,
+                    DevicePixelRatio = 1,
+                },
+            },
+        });
+    // NEXT STATION: Go to Views/Balance.cshtml.
+
+    // Generates a view(Views/History.cshtml) with line chart using <chart> tag helper with two lines to display the monthly amount
+    // of incomes and expenses.
+    // Both of them can be filtered with a tag assigned to it. The tag comes from OC taxonomy.
+    // It uses data from Income and Expense content types stored in db as dataset to it.
+    // Lombiq.ChartJs.Samples/Sample/History
+    public async Task<IActionResult> History(string incomeTag = null, string expenseTag = null)
+    {
+        var incomeTagsFilter = await GetItemIdsByTermIdAsync(ContentItemIds.IncomeTagsTaxonomy, incomeTag);
+        var expenseTagsFilter = await GetItemIdsByTermIdAsync(ContentItemIds.ExpenseTagsTaxonomy, expenseTag);
+
+        var transactions = (await _session.QueryIndex<NumericFieldIndex>(
+            index =>
+                index.Published &&
+                index.Latest &&
+                (index.ContentType == ContentTypes.Income || index.ContentType == ContentTypes.Expense) &&
+                (index.ContentPart == nameof(IncomePart) || index.ContentPart == nameof(ExpensePart)) &&
+                index.ContentField == nameof(TransactionPart.Amount) &&
+                index.Numeric != null)
+            .ListAsync())
+            .Join(
+                await _session.QueryIndex<DateFieldIndex>(
+                index =>
+                    index.Published &&
+                    index.Latest &&
+                    (index.ContentType == ContentTypes.Income || index.ContentType == ContentTypes.Expense) &&
+                    (index.ContentPart == nameof(IncomePart) || index.ContentPart == nameof(ExpensePart)) &&
+                    index.ContentField == nameof(TransactionPart.Date) &&
+                    index.Date != null)
+                .ListAsync(),
+                index => index.ContentItemId,
+                index => index.ContentItemId,
+                (numericIndex, dateIndex) =>
+                    new
+                    {
+                        numericIndex.ContentItemId,
+                        numericIndex.ContentType,
+                        Date = dateIndex.Date.Value,
+                        Amount = numericIndex.Numeric.Value,
```

Comment:
I see `dateIndex.Date` can't be null because it's checked above, so add the `!` to avoid confusion. 
I'm not sure about `dateIndex.Numeric`, so fall back to zero. Generally avoid using plain `Nullable<T>.Value` without `!` becuase it looks like a bug:

![image](https://user-images.githubusercontent.com/4007293/167738329-0ba409d6-603c-43b6-930a-20b833e680c8.png)


```suggestion
                        Date = dateIndex.Date!.Value,
                        Amount = numericIndex.Numeric ?? 0,
```

Reply:
What is the extension you are using for this?

Reply:
I'm using Rider, but if you are on Visual Studio then ReSharper does the same thing. (same engine by the same devs)